### PR TITLE
add naive batching to benchmark.py

### PR DESCRIPTION
**Pull request by xzzWZY**
1. it remembers the "valid" length of each request within a batch, and prints the output for each request after all the requests in the batch finish.
2. it takes total response_length as the sum of all "valid" length within a batch, which is small than the actually output length, since the early stopped request continues to generate some garbage
3. when queries from file reaches the EOF, even if there's not enough requests to fill the batch, it still take the remaining requests in a batch and serve the batched inference, but possibly with a smaller batch size

### CODE REVIEW

File: scripts/benchmark.py [hunk 1]
```diff
@@ -37,6 +39,194 @@
...
+    stopped = np.array(batch_size*[False])
+    for i in range(max_new_tokens):
+        if i == 0:  # prefill
+            if model.config.is_encoder_decoder:
+                out = model.decoder(
+                    input_ids=start_ids,
+                    encoder_hidden_states=encoder_output,
+                    use_cache=True,
+                )
+                logits = model.lm_head(out[0])
```
Comment (jaywonchung) [inline scripts/benchmark.py:108]: Should the `[0]` be here? Have you tested the script for T5?

**Comment by jaywonchung** [comment]
Thank you for your work. The code looks good, but it's actually quite difficult to make sure before running. Can you test vicuna-7B, stablelm, baize-v2, and T5? These are the ones with different `stop_str` and `stop_token_ids` combinations and model architectures.

**Comment by jaywonchung** [comment]
After that, I think we can get it running for A40 now. I'll get the A100 node ready today.

File: scripts/benchmark.py (hunk 1 above)
Comment (xzzWZY) [reply scripts/benchmark.py:108]: yes, it should be there, I've tried on t5

**Comment by xzzWZY** [comment]
wait a minute for merge, I don't test the case when the input file reaches the end, and there's not enough requests to  construct a compete batch; I've just made a toy example for only ten queries and batch 4, and there's some issues; I'll fix it soon

**Comment by jaywonchung** [comment]
No problem, let me know when you've tested the other three models. You can re-request a review on the upper right corner *Reviewers* section.

**Comment by xzzWZY** [comment]
I've test Vicuna-7B and t5 now, it's fine;
by the way when I try to test the models through pegasus, I try to build four new dockers on each gpus, using 

> docker run -dit --name leaderboard_{{gpu}} --gpus '"device={{gpu}}"' -v /data/leaderboard:/data/leaderboard -v $HOME/workspace/leaderboard:/workspace/leaderboard mlenergy/leaderboard:latest bash

since for previous docker like leaderboard0, I cannot git pull to get the lastest version of mlenergy/leaderboard

but there's error message

> docker: Error response from daemon: pull access denied for mlenergy/leaderboard, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.

I've already forward ssh key, and run this out of docker, what's the possible reason(?

**Comment by jaywonchung** [comment]
Can you try again? Now `docker pull mlenergy/leaderboard:latest` and `docker run ... mlenergy/leaderboard:latest bash` should work.

**Comment by xzzWZY** [comment]
it works now, thanks

**Comment by xzzWZY** [comment]
I've just test all four models and all works well

**Comment by xzzWZY** [comment]
I start to run benchmark with batch 4 on 20 models & 4 tasks on ampere02

**Pull request merged**